### ISSUE TITLE
Password UI to access the livemap.

### ISSUE THREAD

**Issue by camonwii** [TGRHavoc/live_map-interface#88]
Is it possible to add a UI in order to add a password to securely access the livemap.

**Comment by TGRHavoc** [comment]
Hey.
In my personal opinion, I think that adding a password shouldn't be implemented as I feel that its out of scope of this project but, I'm willing to have a discussion with you and change my mind if needed.

I have thought about adding this feature in the past but, came to the conclusion that getting it done correctly would be done to the detriment of the user experience.

So, in order for this to be done correctly we would need to do the password processing on the resource side. We could easily do this with a new endpoint (I.e like we do for /blips but, something like /password) but, we cannot guarantee the security of the endpoint. Normally, we would _require_ the password be sent over an encrypted connection (HTTPS) so that third parties cannot snoop on the password and gain access when they shouldn't have. LiveMap isn't something that should need this kind of consideration but, if you're putting a password on LiveMap you're probably doing the same or similar password for a white listed server. So, find one we find the other. Or generally, at least. Not everyone will do this but, it something you have to consider and I just cannot get HTTPS or secure websockets to work.

Secondly, because the whole interface is done on the client side, the user could completely bypass the password UI altogether. Giving no security at all. The only way I can see getting around this is to go back to using PHP which is completely server-side and we can guarantee that users have to enter the password first.

Thirdly, we would also have to put mitigations in place for people that may try and bypass the password by brute forcing etc...

In short, I think its a lot of work for a feature that not many people will use and if they did, I don't think the work would be justified in the outcome.

If someone did want to put LiveMap behind a password, I think there's already a few solutions that already exist. For example, the .htpasswd file that some webservers support.
